docs/source/llms/dspy/notebooks/dspy_quickstart.ipynb

```diff
@@ -0,0 +1,927 @@
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "## 0.3 - Set up DSPy Signature and Module\n",
+    "\n",
+    "Finally, we will define our task: text classification.\n",
+    "\n",
+    "There are a variety of ways you can provide guidelines to DSPy signature behavior. Currently, DSPy allows users to specify:\n",
+    "\n",
+    "1. A high-level goal via the class docstring.\n",
+    "2. A set of input fields, with optional metadata.\n",
+    "3. A set of output fields with optional metadata.\n",
+    "\n",
+    "DSPy will then leverage this information to inform optimization. \n",
+    "\n",
+    "In the below example, note that we **do not** provide any information in our `TextClassificationSignature` class about the dataset of interest. We are effectively leveraging a base LLM instructed to classify text with no context. From this blank slate, we'll look to use DSPy to learn to classify via training alone.\n",
+    "\n",
+    "Note that in production, you should provide metadata in the signature to reduce train time. This secenario is just for demonstration purposes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "d0743c63-c679-4114-b7b1-bed4aeb918cd",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "class TextClassificationSignature(dspy.Signature):\n",
+    "    text = dspy.InputField()\n",
+    "    label = dspy.OutputField(desc=\"Label of predicted_class\")\n",
+    "\n",
+    "\n",
+    "class TextClassifier(dspy.Module):\n",
+    "    def __init__(self):\n",
+    "        super().__init__()\n",
+    "        self.generate_classification = dspy.Predict(TextClassificationSignature)\n",
+    "\n",
+    "    def forward(self, text: str):\n",
+    "        return self.generate_classification(text=text)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "0d93df5c-25d7-460d-a803-0e5469f3bbad",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "# 1 - Run it!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "c89741fa-2698-4409-b290-f5f5652f7d66",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "## 1.1 - Hello World\n",
+    "Let's demonstrate predicting via the DSPy module and associated signature. As expected, the program incorrectly classifies our text because it has not been given the labels."
```

Comment:
Is this cheating? 
Can we at least compare the results between GPT-4 (as you're using) with a basic prompt that includes the label definitions vs. the evaluated results of the compiled and optimized program? 
Are the results compelling in that case?

Reply:
Yes accuracy is still improved with labels. I can refactor.
```
Uncompiled accuracy: 0.625
Compiled accuracy: 0.875
```
For context, the purpose of not providing the labels is to show that DSPy can infer the labels and incorporate them.